lib/activerecord/all/activerecord.rbi

```diff
@@ -226,11 +235,68 @@ module ActiveRecord::Scoping::Named::ClassMethods
 end
 
 class ActiveRecord::Base
-  extend ActiveRecord::Associations::ClassMethods
-  extend ActiveRecord::Inheritance::ClassMethods
-  extend ActiveRecord::Scoping::Named::ClassMethods
-  extend ActiveRecord::NestedAttributes::ClassMethods
-  extend ActiveRecord::Persistence::ClassMethods
+  extend ActiveModel::Naming
```

Comment:
Initially, this list of `include`/`extend` modules was copy-pasted directly: https://github.com/rails/rails/blob/b9ca94cae/activerecord/lib/active_record/base.rb#L278-L325

Anything that didn't come from the rails source I added manually to model transitive concerns/includes/extends. (Those are all commented!)